include/gridtools/common/array.hpp

```diff
@@ -134,7 +134,6 @@ namespace gridtools {
     // in case we need a constexpr version we need to implement a recursive one for c++11
     template <typename T, typename U, size_t D>
     GT_CONSTEXPR GT_FUNCTION bool operator==(array<T, D> const &a, array<U, D> const &b) {
-#pragma unroll
```

Comment:
Do you think we have CUDA code relying on this? Should we run perftests?

Reply:
I don't understand what is wrong with that pragma? What GCC warns about here?

Reply:
That's an nvcc pragma, the gcc pragma would be `#pragma GCC unroll D`

Reply:
Perftests would be good.

I think we can tell the users to do something like this (found on stackoverflow):
```
#pragma GCC diagnostic push 
#pragma GCC diagnostic ignored "-Wunused-local-typedefs"
#include <QtXmlPatterns>
#pragma GCC diagnostic pop
```

Reply:
well, it does not seem to work for me ...

Reply:
Maybe we should just wrap this in an `#ifdef __CUDACC__` then we don't have to do perftests ;)

Reply:
I agree with Hannes about ifdef.
Moreover, we probably should do
```
#ifdef __GNUG__
#pragma GCC unroll
#ifndef __CUDACC__
#pragma unroll  
#endif
```
to be consistent with the approach we take with OMP pragmas

IMHO it doesn't make sense to perftest here. Pragmas can not make the generation worse.

Reply:
The lack of pragmas can make the generation worse

Reply:
Maybe `#idndef _GNUC_` is better, since other compilers also use `#pragma unroll`?